options/docker.go

```diff
@@ -0,0 +1,79 @@
+package options
+
+import (
+	"fmt"
+	"net"
+	"runtime"
+
+	"github.com/docker/docker/client"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/jasper/util"
+	"github.com/pkg/errors"
+)
+
+// Docker encapsulates options related to connecting to a Docker daemon.
+type Docker struct {
+	Host       string
+	Port       int
+	APIVersion string
+	Image      string
+	// Platform refers to the major operating system on which the Docker
+	// container runs.
+	Platform string
+}
+
+// Validate checks whether all the required fields are set and sets defaults if
+// none are specified.
+func (opts *Docker) Validate() error {
+	catcher := grip.NewBasicCatcher()
+	catcher.NewWhen(opts.Port < 0, "port must be positive value")
+	catcher.NewWhen(opts.Image == "", "Docker image must be specified")
+	if opts.Platform == "" {
+		if util.StringSliceContains(DockerPlatforms(), runtime.GOOS) {
+			opts.Platform = runtime.GOOS
+		} else {
+			catcher.Errorf("failed to set default platform to current runtime platform '%s' because it is unsupported", opts.Platform)
+		}
+	} else if !util.StringSliceContains(DockerPlatforms(), opts.Platform) {
+		catcher.Errorf("unrecognized platform '%s'", opts.Platform)
+	}
+	return catcher.Resolve()
+}
+
+// Copy returns a copy of the options for only the exported fields.
+func (opts *Docker) Copy() *Docker {
+	optsCopy := *opts
+	return &optsCopy
+}
+
+// DockerPlatforms returns all supported platforms that can run Docker
+// processes.
+func DockerPlatforms() []string {
```

Comment:
I'd be slightly more inclined to have this be PlatformSupportsDocker(string) bool

Reply:
Done.